micro-rdk-nmea/build.rs

```diff
@@ -0,0 +1,154 @@
+use serde_json::Value;
+use std::fs::{create_dir_all, File, OpenOptions};
+use std::io::{BufReader, Write};
+use std::path::Path;
+
+use convert_case::{Case, Casing};
+
+// This is to compute a valid name for a variable or struct/enum member. We
+// want to remove special characters and introduce a default convention for
+// when a name starts with a number.
+fn clean_string_for_rust(input: &str, counter: usize) -> String {
+    let filtered: String = input
+        .chars()
+        .map(|c| {
+            if c.is_alphanumeric() || c.is_whitespace() {
+                c
+            } else {
+                '_'
+            }
+        })
+        .collect();
+    if filtered
+        .chars()
+        .position(|c| (c.is_numeric() || (c == '_')))
+        .map(|idx| idx == 0)
+        .unwrap_or_default()
+    {
+        // We want to use a prefix with an alphabetical suffix (i.e. "UnformattableVariantA")
+        // when the cleaned string still starts with a number. We map however many variants we've
+        // seen so far to an ASCII code in range 65-90 (A-Z). If there are more than 25 variants
+        // we want to use a second letter (i.e. "UnformattableVariantAB")
+        let suffix = if counter <= 25 {
```

Comment:
So we loose information about what the name  of the thing is right? or can we retrieve it latter? does it even matter?

Reply:
A future PR replaces the number with its word representation instead. This will go away soon. Although I can try to bring the code over in this PR if you prefer